src/Query/Builder.php

```diff
@@ -1053,8 +1053,9 @@ protected function compileWheres(): array
             $method = 'compileWhere' . $where['type'];
             $result = $this->{$method}($where);
 
+            // Negate the expression
```

Comment:
Is `$result` always going to be an `<expression>` at this point, and never an `<operator-expression>` as it expected by [$not](https://www.mongodb.com/docs/manual/reference/operator/query/not/)?

This is related to my other comment, in that I'm curious how this wasn't discovered sooner unless it was only recently introduced.

Reply:
Yes, `compileWhere*` function always return expressions with `$and`, `$or` or a field name, and now `$nor`.